src/filter.js

```diff
@@ -12,15 +11,16 @@ import each from './each'
  */
 
 const filter = (
-  fn: ((mixed) => bool, ?number) => bool,
+  fn: (*, (string | number)) => bool,
   list: mixed[] | Object
 ): Object | mixed[] => {
-  const isArr = isArrayLike(list)
+  const isArr = Array.isArray(list)
```

Comment:
This should be isArrayLike; right now `each` and `filter` also work for strings.